keras/applications/mobilenet.py

```diff
@@ -495,78 +607,3 @@ def _depthwise_conv_block(input, pointwise_conv_filters, alpha,
     x = Activation(relu6, name='conv_pw_%d_relu' % id)(x)
 
     return x
-
-
-def _create_mobilenet(classes, img_input, include_top, alpha, depth_multiplier, dropout, pooling):
```

Comment:
I don't want to delay the merge of this code for this discussion. My comment was hidden in the previous review stage.

As I mentioned in https://github.com/fchollet/keras/pull/7009#discussion_r124160557, the earlier design here would be more effective as part of the public API. I base this assertion on the following research:



- [Best Practices for Scientific Computing. Wilson G et al](http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001745&utm_content=buffer184f2&utm_medium=social&utm_source=twitter.com&utm_campaign=buffer). 
    - [Don't repeat yourself](http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001745&utm_content=buffer184f2&utm_medium=social&utm_source=twitter.com&utm_campaign=buffer#s5)
- [Do Code Clones Matter? Juergens et al](https://arxiv.org/abs/1701.05472)

We should make sure code cloning is not necessary when it is easy, clear, and concise to do so.

I can also provide reference to many repositories where copying of the code is occurring.